src/scripts/main.js

```diff
@@ -2,10 +2,28 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve, reject) => {
+    if (!element) {
+      const error = 'Something went wrong';
```

Comment:
The error message 'Something went wrong' is too generic. Consider providing more specific information about what went wrong, such as indicating that the element is not found or is null.